api/src/Model/Services/UserData.php

```diff
@@ -103,7 +103,7 @@ function getUsers($getCount, $isStaffMember, $stringMatch, $page, $sortBy = null
         $extc = '';
         $group = 'GROUP BY p.personid';
 
-        if (($personId && !$isManager) || $pid || (!$isStaffMember && !$visitName))
+        if ((($personId && !$isManager) || (!$isStaffMember && !$visitName)) && $pid)
         {
             $where .= $this->addPersonOrProposalSearch($pid, $currentUserId, $args);
```

Comment:
Instead this should be 
```suggestion
            $where .= $this->addPersonOrProposalSearch($this->proposalid, $currentUserId, $args);
```

Reply:
When I see the error, if I log $pid and $this->proposalid, they are both blank. This may be because I am staff, but suggests this wouldn't fix the problem.

Reply:
I think it needs to be both of your suggestiosn. If statement should check if $this->proposalid is present and if it is, then pass $this->proposalid instead of $pid because it is such an important part of query its constructing and if it isn't present, then don't attempt to construct that query in the first place. I haven't tested this new change yet.